### DISCUSSION TITLE
Event to trigger workflow when pull request becomes mergeable

### DISCUSSION THREAD

**Discussion by dsager** [community/community#74392] [Actions]
### Select Topic Area

Product Feedback

### Body

We are currently working with the [automerge action](https://github.com/pascalgn/automerge-action) to automatically merge pull requests.
Using branch protection rules we define a couple of restrictions: CI check suite, 2 approvals, no unresolved conversations

This works well except for one downside: The action runs on different triggers (e.g. review submitted, check suite completed) which means it runs many times in vain (because the pull request is not yet mergeable), uselessly consuming GH-action compute minutes.

I wonder if it would make sense to have an event that triggers a workflow as soon as a pull request becomes mergeable, according to the branch-protection rules?

So basically, as soon as a pull request changes from this state:
![image](https://github.com/community/community/assets/1056502/20af3f5f-dd44-4b3f-ab93-3b788bc6c045)

to this state:
![image](https://github.com/community/community/assets/1056502/8777cc39-3fdc-46ae-8d8d-fac6e3ebe1af)

I'd also love to know if there's a simple work around, like checking the pull request's `mergeable` status inside a `jobs.<job_id>.if` conditional?

**Comment by sethicis** [comment]
I just stumbled across your post looking for an answer to the same thing...  Did you ever find a solution for this use case, @dsager ?

**Reply by dsager** [reply]
In the end we use a GH action that enables auto merge for each PR once it is opened & ready for review. Then as soon as all conditions are met, GH merges it. That works quite well.
